src/ApiService/ApiService/UserCredentials.cs

```diff
@@ -9,16 +10,20 @@ namespace Microsoft.OneFuzz.Service;
 public interface IUserCredentials {
     public string? GetBearerToken(HttpRequestData req);
     public string? GetAuthToken(HttpRequestData req);
-    public Task<OneFuzzResult<UserInfo>> ParseJwtToken(HttpRequestData req);
+    public Task<OneFuzzResult<UserAuthInfo>> ParseJwtToken(HttpRequestData req);
 }
 
+public record UserAuthInfo(UserInfo UserInfo, List<string> Roles);
```

Comment:
What if the roles are stored as an enum instead of a string? We could still (de)serialize into a string in requests/responses.

Reply:
Non blocking. just floating the idea

Reply:
The claims are not all defined by us. some of the claims are added by the azure authentication layer. the string type allows us to collect them all